docs/grid_exchange_formats/going_further/datasources.md

```diff
@@ -0,0 +1,158 @@
+(datasources)=
+# Datasources
+
+## Principles
+
+Datasources are Java-objects used to facilitate I/O operations around PowSyBl.
+It allows users to read and write files. It is for example used under the hood by Importers to access the filesystem
+during Network imports when using `Network.read()` methods.
+
+For importers and exporters, datasources are used to access files corresponding to a single network
+
+## Types of datasources
+
+Multiple types of datasources exist, depending on whether it shall be writable or not, the kind of storage used, 
+data location, data compression, etc.
+
+
+(readonlydatasources)=
+### Read-Only DataSource
+
+`ReadOnlyDataSource` is the most basic datasource interface available. As you can tell by the name, it only provides 
+reading features.
+It has two parameters:
+- a base name (corresponding to the common prefix in the names of multiple related files the user wants to 
+consider in the datasource),
+- (optionally) a data extension, mainly used to disambiguate identically named data of different type. 
+Note: this does not apply to compression extensions.
+
+_**Example:**
+For a file named `europe.west.xiidm.gz`, the base name could be `europe.west` for instance (or `europe` or `europe.w` or ...), while the data extension would be `xiidm`._
```

Comment:
```suggestion
For a file named `europe.west.xiidm.gz`, the base name could be `europe.west` for instance (or `europe` or `europe.w` or ...), while the data extension would be `xiidm` and the compression extension `.gz`._
```

Reply:
I prefer to remove the `.gz` in the filename since we are not yet talking about compression